sbtc/src/webhooks.rs

```diff
@@ -20,18 +20,55 @@
 //! Unfortunately, sometimes the same type is serialized using two
 //! different methods for two different parts of the same payload.
 
-use blockstack_lib::burnchains::Txid;
-use blockstack_lib::chainstate::stacks::StacksTransaction;
+use bitcoin::hex::FromHex as _;
 use clarity::vm::types::QualifiedContractIdentifier;
 use clarity::vm::types::Value as ClarityValue;
 use serde::Deserialize;
 use stacks_common::codec::StacksMessageCodec;
 use stacks_common::types::chainstate::BlockHeaderHash;
 use stacks_common::types::chainstate::BurnchainHeaderHash;
 use stacks_common::types::chainstate::StacksBlockId;
-use stacks_common::util::HexDeser;
 
 use crate::error::Error;
+use crate::events::StacksTxid;
+
+/// A trait for deserializing hex encoded strings into the type.
+pub trait DeserializeHex: Sized {
+    /// Try to deserialize the given string into the type.
+    fn try_from_hex(hex: &str) -> Result<Self, Error>;
+}
+
+impl DeserializeHex for StacksTxid {
+    fn try_from_hex(hex: &str) -> Result<Self, Error> {
+        <[u8; 32]>::from_hex(hex)
+            .map(Self)
+            .map_err(Error::DecodeHexTxid)
+    }
+}
+
+impl DeserializeHex for BlockHeaderHash {
+    fn try_from_hex(hex: &str) -> Result<Self, Error> {
+        <[u8; 32]>::from_hex(hex)
```

Comment:
`BlockHeaderHash`, `BurnchainHeaderHash` and `StacksBlockId` have a `from_hex` (via `impl_byte_array_newtype`), we could reuse those (or not, it's not going to look too different either way)